src/uu/comm/src/comm.rs

```diff
@@ -130,7 +130,11 @@ fn open_file(name: &str, line_ending: LineEnding) -> io::Result<LineReader> {
     if name == "-" {
         Ok(LineReader::new(Input::Stdin(stdin()), line_ending))
     } else {
-        let f = File::open(Path::new(name))?;
+        let path = Path::new(name);
+        if metadata(path)?.is_dir() {
+            return Err(io::Error::new(io::ErrorKind::Other, "Is a directory"));
+        }
+        let f = File::open(path)?;
```

Comment:
Is the `path` variable necessary? Unless I'm missing something I think you can pass `name` directly to `metadata` and `File::open`.